ansible/roles/atmo-user-deployment-scripts/tasks/main.yml

```diff
@@ -58,38 +58,23 @@
     state: link
   with_items: '{{ ASYNC_SCRIPTS }}'
 
-#Future TODO:
-# - If possible, refactor these components into ansible:
-#   - get name of  each file in script directory
-#   - if file is executable, generate 'dynamic log files' based on 'script|basename' and 'date()'
-#   - if file is executable, run ` set -m; nohup-call-to-file..` and point to dynamic log files from above.
-- name: deploy_script_6b - execute scripts found in ASYNC_RUN_SCRIPTS asynchronously
-  shell: |
-    set -m  # allows backgrounded processes to persist
-    for SCRIPT in {{item.dir}}/*;
-    do
-      if [[ -f $SCRIPT && -x $SCRIPT ]]; then
-        LOG_DIR="{{item.log}}"
-        BASENAME=${SCRIPT##*/}
-        DATE=$(date +%F_%T)
-        SCRIPT_STDOUT=$LOG_DIR/$BASENAME.$DATE.stdout
-        SCRIPT_STDERR=$LOG_DIR/$BASENAME.$DATE.stderr
-        ATMO_USER={{ATMOUSERNAME}} nohup $SCRIPT >$SCRIPT_STDOUT 2>$SCRIPT_STDERR </dev/null &
-      fi
-    done
-    exit 0
-  args:
-    executable: /bin/bash
-  register: shell_result
-  with_items: '{{ ASYNC_RUN_SCRIPTS }}'
+- name: deploy_script_6a - Execute scripts
+  shell: 'ATMO_USER={{ ATMOUSERNAME }} nohup {{ INSTANCE_SCRIPT_DIR }}/{{ item.name }} &'
+  register: SCRIPT_RESULT
+  with_items: '{{ ASYNC_SCRIPTS }}'
+
+- name: deploy_script_6b - Register date
+  command: 'date +"%Y-%m-%d-%H:%M"'
+  register: DATE
 
-# Helpful if the shell script above fails..
-# - debug: msg="item.item={{item.item}}, item.stdout={{item.stdout}}, item.changed={{item.changed}}"
-#   with_items: "{{shell_result.results}}"
+- name: deploy_script_6c - Create logs
+  template:
+    src: 'deploy_log.j2'
+    dest: '{{ INSTANCE_SCRIPT_LOG_DIR }}/{{ item.item.name }}_{{ DATE.stdout }}.log'
```

Comment:
Should this be item.name?

Reply:
Nope. The variable `SCRIPT_RESULT` is registered from a task with a loop, so each pass of the loop is stored in the list `SCRIPT_RESULT.results`. Then each item in `results` has an `item` attribute that is the item from `ASYNC_SCRIPTS` that was used.

Reply:
I'm not sure if I was very clear above so here is an excerpt of `SCRIPT_RESULT` variable. I cut it off after the first entry in `results`:

```
{
    "msg": "All items completed", 
    "changed": true, 
    "results": [
        {
            "_ansible_parsed": true, 
            "stderr_lines": [], 
            "changed": true, 
            "stdout": "", 
            "_ansible_item_result": true, 
            "delta": "0:00:01.002607", 
            "stdout_lines": [], 
            "end": "2018-04-18 11:46:44.427359", 
            "_ansible_no_log": false, 
            "start": "2018-04-18 11:46:43.424752", 
            "cmd": "ATMO_USER=calvinmclean nohup /etc/atmo/instance-scripts.d/async1 &", 
            "item": {
                "text": "#!/bin/bash\nsleep 10\necho \"async1 - 1\" >> /etc/script-test\nsleep 3\necho \"async1 - 2\" >> /etc/script-test", 
                "name": "async1"
            }, 
            "stderr": "", 
            "rc": 0, 
            "invocation": {
                "module_args": {
                    "warn": true, 
                    "executable": null, 
                    "_uses_shell": true, 
                    "_raw_params": "ATMO_USER=calvinmclean nohup /etc/atmo/instance-scripts.d/async1 &", 
                    "removes": null, 
                    "creates": null, 
                    "chdir": null
                }
            }
        }, 
```